File: Тор Academy/1. Основы IT/ДЗ - 7/Задание1.cpp
#include <iostream>

long sum_even_fibonacci(long limit = 4000000) {

    long num1 = 0, num2 = 1, result_sum = 0;

    while (true) {

        long temp = num1;
        num1 = num2;
        num2 = temp + num2;

        if (num2 % 2 == 0) result_sum += num2;
        
        if (num2 + num1 > limit) break;
    
    }
    return result_sum;
}

int main()
{
    using namespace std;
    setlocale(LC_ALL, "RU");

    cout << "Ответ: " << sum_even_fibonacci() << endl;
    

}

File: Тор Academy/1. Основы IT/ДЗ - 7/Задание1.py
'''Найти сумму всех четных элементов рада Фибоначи
не превышающих 4млн.
'''


def sum_even_fibonaci(limit=4_000_000):

    num1 = num2 = 1
    result = 0

    while True:

        num1, num2 = num2, num1 + num2

        if not num2 % 2:
            result += num2

        if num1 + num2 > limit:
            break

    return result


print(sum_even_fibonaci())






















'''c Решение 2 (через цикл for)'''

# def fibonacci_barometr_2(generation: int) -> int:
#     num1 = num2 = 1
    
#     for i in range(2, generation):
#         num1, num2 = num2, num1 + num2
#         print(num2, end = " ")
    
# fibonacci_barometr_2(6)